include/envoy/http/codec.h

```diff
@@ -386,6 +392,12 @@ struct Http1Settings {
 
   // How header keys should be formatted when serializing HTTP/1.1 headers.
   HeaderKeyFormat header_key_format_{HeaderKeyFormat::Default};
+
+  // Behaviour on invalid HTTP messaging:
+  // - if set and true, the HTTP/1.1 connection is left open (where possible)
+  // - if set and false, the HTTP/1.1 connection is terminated
+  // - if not set, fall back to the HCM behaviour on invalid HTTP messaging
```

Comment:
Similar comment here about HCM. This is generic for upstream and downstream. Where does the upstream default come from?

Reply:
The new HTTP/1.1 config introduced here works in tandem with the newly introduced `stream_override_on_invalid_http_message` API for the HCM, and it's a noop when used in the context of an upstream where there is no HCM. Maybe this should be better clarified in the comments?!

Reply:
Yeah I think it would be good to call this out more in the comments and the proto documentation. The proto is a generic message that is used for both upstream and downstream, so I think it would be good to clarify that it only applies right now for downstream?